server.js

```diff
@@ -1,30 +1,64 @@
-import express from 'express'
+import express, { response } from 'express'
 import bodyParser from 'body-parser'
 import cors from 'cors'
 
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
-
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
+import booksData from './data/books.json'
+
 const port = process.env.PORT || 8080
 const app = express()
 
-// Add middlewares to enable cors and json body parsing
+const BOOKS_NOT_FOUND = { error: 'No books where found'} 
 app.use(cors())
 app.use(bodyParser.json())
 
-// Start defining your routes here
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send('Hi and welcome to my book review site. To get the full list use the endpoint /books')
+})
+
+app.get('/books', (req, res) => {
+  res.json(booksData)
+})
+
+app.get('/books/:id', (req, res) => {
+  const bookId = req.params.id
+  const book = booksData.find(book => book.bookID === +bookId)
+  if (!book) {
+    res.status(404).json(BOOKS_NOT_FOUND)
+  }
+    res.json(book)
+})
+
+app.get('/authors/:author', (req, res) => {
+  const { author } = req.params
+  const byAuthor = booksData.filter(book => book.authors === author)
+  if (!byAuthor.length) {
+    res.status(404).json(BOOKS_NOT_FOUND)
+  }
+    res.json(byAuthor)
+})
+
+app.get('/title/:title', (req, res) => {
```

Comment:
you filter the book list by title, not the list of titles, so it should be `/books/title/:title`